bigquery/google/cloud/bigquery/job.py

```diff
@@ -1384,20 +1368,39 @@ def from_api_repr(cls, resource):
     https://g.co/cloud/bigquery/docs/reference/rest/v2/jobs#configuration.query.createDisposition
     """
 
-    default_dataset = _TypedApiResourceProperty(
-        'default_dataset', 'defaultDataset', DatasetReference)
-    """See
-    https://g.co/cloud/bigquery/docs/reference/v2/jobs#configuration.query.defaultDataset
-    """
+    @property
+    def default_dataset(self):
+        """google.cloud.bigquery.dataset.DatasetReference: the default dataset
+        to use for unqualified table names in the query.
 
-    destination = _TypedApiResourceProperty(
-        'destination', 'destinationTable', TableReference)
-    """
-    google.cloud.bigquery.table.TableReference: table where results are written
+        See
+        https://g.co/cloud/bigquery/docs/reference/v2/jobs#configuration.query.defaultDataset
+        """
+        prop = self._properties.get('defaultDataset')
+        if prop is not None:
```

Comment:
The docstring should indicate that the value could possibly be None if unset.

Maybe:

```
google.cloud.bigquery.dataset.DatasetReference: a reference to
the default dataset ... or ``None`` if not set.
```

Same for the other properties, like `destination`.

Reply:
Agreed. Added in 55fad6c